zebra-chain/src/transaction/serialize.rs

```diff
@@ -197,6 +198,28 @@ impl ZcashSerialize for Transaction {
                 inputs.zcash_serialize(&mut writer)?;
                 outputs.zcash_serialize(&mut writer)?;
 
+                match sapling_shielded_data {
+                    None => {
+                        // Signal no value balance.
+                        writer.write_i64::<LittleEndian>(0)?;
+                        // Signal no shielded spends and no shielded outputs.
+                        writer.write_compactsize(0)?;
+                        writer.write_compactsize(0)?;
```

Comment:
Transaction v5 has more arrays than this.

Please reference the relevant spec section in a comment.